### DISCUSSION TITLE
resilient.tableofcontents: toc-level1 runs of the typeblock and vbreak has no effect.

### DISCUSSION THREAD

**Discussion by no-vici** [Omikhleia/resilient.sile#20] [Q&A] [answered]
How can I force line-break for the tableofcontents?

![resilient_style_tableofcontent](https://user-images.githubusercontent.com/113570767/225889396-2cded62f-adab-4053-bd63-30c3becaeb9d.png)

**Comment by Omikhleia** [comment, marked as answer]
That's a bug actually, due to the fact that PDF links (in which we wrap ToC entries) cannot be line-broken yet in SILE...

https://github.com/sile-typesetter/sile/issues/1239 (tentative PR https://github.com/sile-typesetter/sile/pull/1334)

So currently, the ToC line _will_ overflow the page as observed, if PDF links are enabled. 

A fix for this issue is not fully sufficient (we'd still have to _reserve_ the space for the page number, i.e. to break line at an earlier point that the frame edge), but for now, until it gets a solution, I've refrained considering it.

In other terms, disabling PDF linking (`\tableofcontents[linking=true]`) will somewhat work, but is not perfect either.

**Reply by Omikhleia** [reply]
> In other terms, disabling PDF linking (`\tableofcontents[linking=true]`) 

Oops. I mean `\tableofcontents[linking=false]`, obviously.

**Reply by Omikhleia** [reply]
>  but for now, until it gets a solution, I've refrained considering it.

... And on a second thought, I had a bit forgotten, but last time I checked this, I hadn't yet the "minimization" feature in the **parbox** package, which is the trick I used in the **fancytoc** package. Hmm... Perhaps it could be used here too. It's a bit of a heavy-handed workaround (akin to using a table cell to solve the problem), but maybe it would be acceptable...

But it mostly depends on where people would expect the line break to occur and the dots + page number to be shown... I mean, what would be the most logical expectations here (image hacked in Gimp just for the purpose of illustration):

![image](https://user-images.githubusercontent.com/18075640/225907326-66af6445-4c27-48ff-aa05-019d9e74317e.png)

My strategy in **fancytoc** is the third one, i.e. the dots + page no on the first line, and the second line wrapped with an indent. It was easy, but there we do not show any (part / chapter / section) numbers, which would make the "indent" not so obvious to do right.

**Reply by Omikhleia** [reply]
Oh interesting article on a similar topic in (La)TeX = https://tug.org/TUGboat/tb33-3/tb105veytsman-leaders.pdf

**Reply by no-vici** [reply]
> Oops. I mean `\tableofcontents[linking=false]`, obviously.

Yep, I've realised ;) Thank you.

**Reply by no-vici** [reply]
> > but for now, until it gets a solution, I've refrained considering it.
> 
> ... And on a second thought, I had a bit forgotten, but last time I checked this, I hadn't yet the "minimization" feature in the **parbox** package, which is the trick I used in the **fancytoc** package. Hmm... Perhaps it could be used here too. It's a bit of a heavy-handed workaround (akin to using a table cell to solve the problem), but maybe it would be acceptable...

Let me try to change the font-size of the toc-level to see if the text fixed within the main text frame.

> But it mostly depends on where people would expect the line break to occur and the dots + page number to be shown... I mean, what would be the most logical expectations here (image hacked in Gimp just for the purpose of illustration):

Would it look nicer that the page number sits on the same line with the second line of the broken text?

**Reply by no-vici** [reply]
>  Oops. I mean `\tableofcontents[linking=false]`, obviously.

This trick works ;) I also reduced the font-size of the toc-level(N) and they fit nicely now. 

Thank you!

**Comment by Omikhleia** [comment]
**For the record**, note that since [version 2.3.0](https://github.com/Omikhleia/resilient.sile/releases/tag/v2.3.0), _re·sil·ient_ supports multi-line links (and this became standard too in SILE 0.15), so the above "workaround" disabling links in the ToC should not be necessary any longer, i.e. one would now get line-breaks instead of an entry running "off the typeblock".

(The more general discussion on _how_ multi-line ToC entries should eventually show is not addressed fully)

**Reply by no-vici** [reply]
It sounds great.
> (The more general discussion on how multi-line ToC entries should eventually show is not addressed fully)
means that at the user level, I don't need to know how it works ;)

I will mess around to see how this new feature rolls out.
